include/oneapi/dpl/experimental/kt/esimd_radix_sort_utils.h

```diff
@@ -66,28 +66,109 @@ gather(const T* input, sycl::ext::intel::esimd::simd<::std::uint32_t, N> offsets
+
+        __uint32_mask[2 * src_index    ] = mask[src_index];
+        __uint32_mask[2 * src_index + 1] = mask[src_index];
+    }
+
+    sycl::ext::intel::esimd::simd<uint32_t, 2 * N> __uint32__result = utils::gather(*__p_uint32_acc, __uint32_offsets, base_offset, __uint32_mask);
+    return __uint32__result.template bit_cast_view<T>();
 }
 
 template <typename T, int N>
 void
 scatter(T* output, sycl::ext::intel::esimd::simd<::std::uint32_t, N> offsets,
-        sycl::ext::intel::esimd::simd<T, N> values, sycl::ext::intel::esimd::simd_mask<N> mask = 1)
+        sycl::ext::intel::esimd::simd<T, N> values,
+        sycl::ext::intel::esimd::simd_mask<N> mask = 1)
 {
     sycl::ext::intel::esimd::scatter(output, offsets * size32<T>, values, mask);
 }
 
-template<typename T, int N, sycl::access_mode Mode, sycl::access::placeholder P>
+template<typename T, int N, sycl::access_mode Mode, sycl::access::placeholder P, std::enable_if_t<sizeof(T) <= 4, int> = 0>
 void
-scatter(sycl::accessor<T, 1, Mode, sycl::target::device, P> output, sycl::ext::intel::esimd::simd<::std::uint32_t, N> offsets,
-        sycl::ext::intel::esimd::simd<T, N> values, sycl::ext::intel::esimd::simd_mask<N> mask = 1)
+scatter(sycl::accessor<T, 1, Mode, sycl::target::device, P> acc,
+        sycl::ext::intel::esimd::simd<::std::uint32_t, N> offsets,
+        sycl::ext::intel::esimd::simd<T, N> values,
+        sycl::ext::intel::esimd::simd_mask<N> mask = 1)
 {
-    sycl::ext::intel::esimd::scatter(output, offsets * size32<T>, values, /*global_offset*/ 0, mask);
+    // https://intel.github.io/llvm-docs/doxygen/group__sycl__esimd__memory.html#ga851442aaf354eebfda03800487bb1091
+    // template<typename T , int N, typename AccessorTy , typename Toffset >
+    // __ESIMD_API
+    // std::enable_if_t<(sizeof(T) <= 4) && (N == 1 || N == 8 || N == 16 || N == 32) && !std::is_pointer<AccessorTy>::value && std::is_integral_v<Toffset>>
+    // sycl::_V1::ext::intel::esimd::scatter(AccessorTy acc,
+    //                                       simd<Toffset, N> offsets,
+    //                                       simd<T, N> vals,
+    //                                       uint32_t glob_offset = 0,
+    //                                       simd_mask<N> mask = 1);
+
+    sycl::ext::intel::esimd::scatter(acc, offsets * size32<T>, values, /*global_offset*/ 0, mask);
+}
+
+template<typename T, int N, sycl::access_mode Mode, sycl::access::placeholder P, std::enable_if_t<sizeof(T) == 2 * sizeof(uint32_t), int> = 0>
+void
+scatter(sycl::accessor<T, 1, Mode, sycl::target::device, P> acc,
+        sycl::ext::intel::esimd::simd<::std::uint32_t, N> offsets,
+        sycl::ext::intel::esimd::simd<T, N> values,
+        sycl::ext::intel::esimd::simd_mask<N> mask = 1)
+{
+    // TODO is it correct?
+    using uint32Acc = sycl::accessor<uint32_t, 1, Mode, sycl::target::device, P>;
+
+    uint32Acc*                                            __p_uint32_acc = reinterpret_cast<uint32Acc*>(&acc);
```

Comment:
@dmitriy-sobolev what do you think about this ```reinterpret_cast``` conversion ?

Reply:
I left my opinion here: https://github.com/oneapi-src/oneDPL/pull/968/files#r1210088317